tools/cli/domainCommands.go

```diff
@@ -901,14 +913,68 @@ func (d *domainCLIImpl) migrationDynamicConfigCheck(c *cli.Context) DomainMigrat
 
 			if currResp.Value != newResp.Value {
 				check = false
-				mismatchedConfigs = append(mismatchedConfigs, MismatchedDynamicConfig{
-					CurrResp: currResp,
-					NewResp:  newResp,
+				mismatchedConfig = MismatchedDynamicConfig{
+					Key: configKey,
+					CurrValues: []*types.DynamicConfigValue{
+						toDynamicConfigValue(currResp.Value, map[dc.Filter]interface{}{
+							dynamicconfig.DomainID: currentDomainID,
+						}),
+					},
+					NewValues: []*types.DynamicConfigValue{
+						toDynamicConfigValue(currResp.Value, map[dc.Filter]interface{}{
+							dynamicconfig.DomainID: destinationDomainID,
+						}),
+					},
+				}
+			}
+
+		} else if containsFilter(configKey, dc.DomainName.String()) && containsFilter(configKey, dc.TaskListName.String()) {
+			taskLists := c.StringSlice(FlagTaskList)
+			for _, taskList := range taskLists {
```

Comment:
aggregate all values into single mismatchedConfig
```suggestion
			taskLists := c.StringSlice(FlagTaskList)
			var mismatchedCurValues, mismatchedNewValues []*types.DynamicConfigValue]
			for _, taskList := range taskLists {
```